src/main/java/se/sundsvall/casemanagement/api/validation/impl/OneMainFacilityOrNullConstraintValidator.java

```diff
@@ -0,0 +1,21 @@
+package se.sundsvall.casemanagement.api.validation.impl;
+
+import java.util.List;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+import se.sundsvall.casemanagement.api.model.FacilityDTO;
+import se.sundsvall.casemanagement.api.validation.OneMainFacilityOrNull;
+
+public class OneMainFacilityOrNullConstraintValidator implements ConstraintValidator<OneMainFacilityOrNull, List<FacilityDTO>> {
+
+	@Override
+	public boolean isValid(List<FacilityDTO> planningPermissionFacilityList, ConstraintValidatorContext context) {
+		if (planningPermissionFacilityList == null) {
+			// We can't do this validation if the list is null
+			return true;
+		}
+		return planningPermissionFacilityList.stream().filter(FacilityDTO::isMainFacility).count() <= 1;
```

Comment:
The validator is named `OneMainFacilityOrNullConstraintValidator`. Shouldn't count be equal to 1 here for the input to be valid? If no facility is main facility this validatror will return true, is that correct?

If the validator is verifying that no more than one facility is main, maybe rename validator to MaxOneMainFacilityValidator?